### ISSUE TITLE
Zope Plugin breaks contextmanager plugin

### ISSUE THREAD

**Issue by euresti** [Shoobx/mypy-zope#35]
Here's an example:
```
from collections import Iterator
from contextlib import contextmanager
from typing import TypeVar, Generic

_T = TypeVar("_T")


class A(Generic[_T]):
    pass

@contextmanager
def m(x: _T) -> Iterator[A[_T]]:
    ...


with m(7) as x:
    reveal_type(x)
    print(x)
```

Without the plugin you get: `note: Revealed type is 'bar.A*[builtins.int*]'`
With the plugin you get: `Revealed type is 'bar.A*[_T`-1]'`

I tracked the issue down to `get_function_hook` doing a `return analyze` for all methods thus preventing the hooks at `mypy.plugins.default.DefaultPlugin.get_function_hook` from running.  

This is the same as #16 

Looking at https://github.com/python/mypy/issues/8331 I tried figuring out how one could use `lookup_fully_qualified` and here's some code that works for some cases. 

```
        info = self.lookup_fully_qualified(fullname)
        if (
            info and info.node and isinstance(info.node, SYMBOL_FUNCBASE_TYPES)
            and info.node.type and isinstance(info.node.type, CallableType)
            and info.node.type.ret_type and isinstance(info.node.type.ret_type, Instance)
        ):
            info.node = info.node.type.ret_type.type

        if (info
            and info.node
            and isinstance(info.node, TypeInfo)
                and info.node.mro
                and any(typeinfo.fullname in {"zope.interface.interface.Attribute", "zope.schema.fieldproperty.FieldProperty"}
               for typeinfo in self.lookup_fully_qualified(fullname).node.mro)):
            return analyze
```

**Comment by kedder** [comment]
@euresti, I see the problem, thanks for debugging it. It seems like this plugin chaining mechanism of mypy is rather limited - I couldn't find a way to fallback to the DefaultPlugin once you "claim" the function hook.

Of course, we do it ourselves by explicitly calling DefaultPlugin inside the `analize()` function. But obviously this will only really work if mypy_zope is the only other plugin in mypy's plugin chain. I guess we'll have to accept this drawback for the time being.

I'll try to come up with the MR.

**Comment by kedder** [comment]
@euresti this should be fixed by #36. Do you want a new plugin release?

**Comment by euresti** [comment]
Looks like this fixes my current issue.  A new plugin release would be nice.  Thanks!

**Comment by kedder** [comment]
@euresti, released 0.2.12 with the fix.
